eth_account/messages.py

```diff
@@ -14,14 +14,23 @@
 )
 
 
-def defunct_hash_message(primitive=None, hexstr=None, text=None):
+def defunct_hash_message(
+        *,
+        primitive=None,
+        hexstr=None,
+        text=None,
+        signature_version=b'E',
+        version_specific_data=None):
     '''
     Convert the provided message into a message hash, to be signed.
     This provides the same prefix and hashing approach as
-    :meth:`w3.eth.sign() <web3.eth.Eth.sign>`. That means that the
-    message will automatically be prepended with text
-    defined in EIP-191 as version 'E': ``b'\\x19Ethereum Signed Message:\\n'``
+    :meth:`w3.eth.sign() <web3.eth.Eth.sign>`. The message will by default
+    be prepended with text defined in EIP-191 as
+    version 'E': ``b'\\x19Ethereum Signed Message:\\n'``
     concatenated with the number of bytes in the message.
+    Currently you can only specify the ``signature_version`` as
+    version ``0``: Sign data with intended validator (EIP 191).
+    Here the version_specific_data would be a hexstr which is the 20 bytes account address
 
     Awkwardly, the number of bytes in the message is encoded in decimal ascii. So
```

Comment:
I think this paragraph would be better right after we talk about version E, since it only applies there. What do you think?

Reply:
@kclowes how about we first mention all the `versions` we are supporting, and then we talk about this case with mention of `version E` specifically. Just for having all of them at the same place.